zetaclient/evm/inbounds_test.go

```diff
@@ -333,3 +355,59 @@ func TestEVM_BuildInboundVoteMsgForTokenSentToTSS(t *testing.T) {
 		require.Nil(t, msg)
 	})
 }
+
+func TestEVM_ObserveTSSReceiveInBlock(t *testing.T) {
+	// https://etherscan.io/tx/0xeaec67d5dd5d85f27b21bef83e01cbdf59154fd793ea7a22c297f7c3a722c532
+	chain := common.EthChain()
+	chainID := chain.ChainId
+	confirmation := uint64(1)
+	chainParam := stub.MockChainParams(chain.ChainId, confirmation)
+	intxHash := "0xeaec67d5dd5d85f27b21bef83e01cbdf59154fd793ea7a22c297f7c3a722c532"
+
+	// load archived tx and receipt
+	tx, receipt := testutils.LoadEVMIntxNReceipt(t, chainID, intxHash, common.CoinType_Gas)
+	require.NoError(t, evm.ValidateEvmTransaction(tx))
+
+	// load archived evm block
+	// https://etherscan.io/block/19363323
+	blockNumber := receipt.BlockNumber.Uint64()
+	block := testutils.LoadEVMBlock(t, chainID, blockNumber, true)
+
+	// create mock client
+	evmClient := stub.NewMockEvmClient()
+	evmJSONRPC := stub.NewMockJSONRPCClient()
+	zetaClient := stub.NewMockZetaCoreBridge()
+	tss := stub.NewTSSMainnet()
+	lastBlock := receipt.BlockNumber.Uint64() + confirmation
+
+	t.Run("should observe TSS receive in block", func(t *testing.T) {
+		ob := MockEVMClient(chain, evmClient, evmJSONRPC, zetaClient, tss, lastBlock, chainParam)
+
+		// feed archived block and receipt
+		evmJSONRPC.WithBlock(block)
+		evmClient.WithReceipt(receipt)
+		err := ob.ObserveTSSReceiveInBlock(blockNumber)
+		require.NoError(t, err)
+	})
+	t.Run("should not observe on error getting block", func(t *testing.T) {
+		ob := MockEVMClient(chain, evmClient, evmJSONRPC, zetaClient, tss, lastBlock, chainParam)
+		err := ob.ObserveTSSReceiveInBlock(blockNumber)
+		require.ErrorContains(t, err, "error getting block")
+	})
+	t.Run("should not observe on error getting receipt", func(t *testing.T) {
+		ob := MockEVMClient(chain, evmClient, evmJSONRPC, zetaClient, tss, lastBlock, chainParam)
+		evmJSONRPC.WithBlock(block)
+		err := ob.ObserveTSSReceiveInBlock(blockNumber)
+		require.ErrorContains(t, err, "error getting receipt")
+	})
+	t.Run("should not observe on error posting vote", func(t *testing.T) {
+		ob := MockEVMClient(chain, evmClient, evmJSONRPC, zetaClient, tss, lastBlock, chainParam)
+
+		// feed archived block and pause zeta bridge
+		evmJSONRPC.WithBlock(block)
+		evmClient.WithReceipt(receipt)
+		zetaClient.Pause()
+		err := ob.ObserveTSSReceiveInBlock(blockNumber)
+		require.ErrorContains(t, err, "error checking and voting")
```

Comment:
Just for the sake of maintainability.
Can we add a comment, why these assertions are expected in the test case?
Example: 
```
// error getting block is expected because the call has not been mocked with "WithBlock"
```

Reply:
https://github.com/zeta-chain/node/pull/1878/commits/4ae7c1692cc6a35d4f7d940722c4e72521a67277